### PR TITLE
Moths can now eat clothes

### PR DESCRIPTION
**Pull request by Mothblocks**
[Changelogs]:

:cl:
add: Moths can now eat clothes.
/:cl:

[why]: I thought it sounded funny and really just wanted to see if I could code something abstract like this. Provides extremely little nutritional value, it's just funny.

Apparently Hippie made this exact change as a PR 3 days ago. I didn't know but I also don't care.

### CODE REVIEW

File: code/modules/clothing/clothing.dm [hunk 1]
```diff
@@ -56,6 +56,25 @@
...
+	name = "oops"
+	desc = "If you're reading this it means I messed up. This is related to moths eating clothes and I didn't know a better way to do it than making a new food object."
+	list_reagents = list("nutriment" = 1)
+	tastes = list("dust" = 1, "lint" = 1)
+
+/obj/item/clothing/attack(mob/M, mob/user, def_zone)
+	if(user.a_intent != INTENT_HARM && ismoth(M))
+		var/obj/item/reagent_containers/food/snacks/clothing/clothing_as_food = new
+		clothing_as_food.name = name
+		//clothing_as_food.attack(M, user, def_zone) <--- this errored and said some shit about types
```
Comment (XDTM) [inline code/modules/clothing/clothing.dm]: Maybe it needs to be `M.attack(clothing_as_food, etc)` ?

Comment (Jared-Fogle) [reply code/modules/clothing/clothing.dm]: attack is a member of /obj/item though which is the weird part?

Comment (Jared-Fogle) [reply code/modules/clothing/clothing.dm]: Actually wait I think this was fixed when I actually defined /food/snacks/clothing, recompiling to see if it fixed itself.

Comment (Jared-Fogle) [reply code/modules/clothing/clothing.dm]: Yeah it was, I'll test then push.

**Comment by challade** [comment]
Hey Jared I notice you and I both had the same idea in coding for moths to eat clothing, and that you don't have a profile pic. You should totally use the one I provided, it'd be funny. Your choice though. Let's cross our fingers to see if either code works. Good luck brother.
pull

**Comment by JStheguy** [comment]
@Time-Green 
Well, to do that you'd probably have to also component-ify reagent_storage as a whole, so good luck to the brave spaceman who tries.

**Comment by Qustinnus** [comment]
@challade thanks

File: code/modules/clothing/clothing.dm [hunk 2]
```diff
@@ -56,6 +56,24 @@
 		if(M.putItemFromInventoryInHandIfPossible(src, H.held_index))
 			add_fingerprint(usr)
 
+/obj/item/reagent_containers/food/snacks/clothing
+	name = "oops"
+	desc = "If you're reading this it means I messed up. This is related to moths eating clothes and I didn't know a better way to do it than making a new food object."
+	list_reagents = list("nutriment" = 1)
+	tastes = list("dust" = 1, "lint" = 1)
+
+/obj/item/clothing/attack(mob/M, mob/user, def_zone)
```
Comment (optimumtact) [inline code/modules/clothing/clothing.dm]: this is incredibly low kino code

File: code/modules/clothing/clothing.dm [hunk 3]
```diff
@@ -56,6 +56,24 @@
...
+	name = "oops"
+	desc = "If you're reading this it means I messed up. This is related to moths eating clothes and I didn't know a better way to do it than making a new food object."
+	list_reagents = list("nutriment" = 1)
+	tastes = list("dust" = 1, "lint" = 1)
+
+/obj/item/clothing/attack(mob/M, mob/user, def_zone)
+	if(user.a_intent != INTENT_HARM && ismoth(M))
+		var/obj/item/reagent_containers/food/snacks/clothing/clothing_as_food = new
+		clothing_as_food.name = name
+		var/eaten = clothing_as_food.attack(M, user, def_zone)
```
Comment (nicbn) [inline code/modules/clothing/clothing.dm]:
```
if(clothing_as_food.attack(M, user, def_zone))
	take_damage(15)
qdel(clothing_as_food)
```

Comment (Jared-Fogle) [reply code/modules/clothing/clothing.dm]: I can fix that in a bit

Comment (nicbn) [reply code/modules/clothing/clothing.dm]: It's not exactly a fix as it's not a bug, it's just removal of a redundant variable

Comment (Jared-Fogle) [reply code/modules/clothing/clothing.dm]: That's what I meant

**Comment by Jared-Fogle** [comment]
Just realized there's a loud attack sound when you eat clothes. Fixing it.

**Comment by Durkel** [comment]
Moths don't eat clothing, most don't even have any form of chewing mechanism and subsist on drinking water or nectar. It's the larvae that eat clothing.

**Comment by Jared-Fogle** [comment]
I'm not exactly going for lore or accuracy here.

**Comment by Jared-Fogle** [comment]
@Bawhoppen Because eating clothes is something people relate to moths, regardless of whether it's only the larvae or not

**Pull request merged**